src/huggingface_hub/hf_api.py

```diff
@@ -499,20 +498,34 @@ def _is_valid_token(self, token: str):
         except HTTPError:
             return False
 
-    def _validate_or_retrieve_token(self, token: Optional[Union[str, bool]] = None):
-        """
-        Either retrieves stored token or validates passed token.
-        """
+    def _validate_or_retrieve_token(
+        self,
+        token: Optional[str] = None,
+        name: Optional[str] = None,
+        function_name: Optional[str] = None,
+    ):
         if token is None or token is True:
             token = HfFolder.get_token()
             if token is None:
                 raise EnvironmentError(
                     "You need to provide a `token` or be logged in to Hugging Face with "
                     "`huggingface-cli login`."
                 )
-        elif not self._is_valid_token(token):
-            raise ValueError("Invalid token passed!")
-        return token
+        if name is not None:
+            if self._is_valid_token(name):
+                warnings.warn(
```

Comment:
this needs to be removed in the future.

```suggestion
                # TODO(0.6) REMOVE
                warnings.warn(
```